src/imitation/algorithms/dagger.py

```diff
@@ -310,25 +358,39 @@ def _check_has_latest_demos(self):
                 f".get_trajectory_collector()"
             )
 
-    def _try_load_demos(self):
-        self._check_has_latest_demos()
         if self._last_loaded_round < self.round_num:
+            # TODO(shwang): Do we actually only want to load the most recent round's
+            #   data, or should we load all the data?
```

Comment:
@qxcv: IIRC in DAgger we want to update BC on the entire aggregated dataset. However, it looks like we only load the most recent round's data here. Is this intentional?

If this is indeed the intended behvario, then we should change `self._last_loaded_round < self.round_num`  to `self._last_loaded_round != self.round_num`.

Reply:
Yes, the intent is to load demos from all rounds.

Reply:
Thanks, I was confused there but that's right